core/src/main/java/org/bitcoinj/core/TransactionOutput.java

```diff
@@ -169,14 +169,14 @@ public Coin getValue() {
     }
 
     /**
-     * Sets the value of this output.
+     * Returns a clone of this output, with given value. The typical use case is fee calculation.
+     *
+     * @param value value for the clone
+     * @return clone of output, with given value
      */
-    public void setValue(Coin value) {
+    public TransactionOutput withValue(Coin value) {
         Objects.requireNonNull(value);
```

Comment:
non-null `value` is checked (implicitly) by the called constructor, so this could be removed.

Reply:
True, but isn't it convenient if the called method is on top of the stack trace?

Reply:
I wonder if the JIT is smart enough to figure out that this only needs to be done once in that use case?

Reply:
I'd assume no, but is that really relevant for performance? It's just a null check.